src/main/kotlin/org/rust/ide/template/postfix/PrintlnPostfixTemplate.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.template.postfix
+
+import com.intellij.codeInsight.template.postfix.templates.PostfixTemplateWithExpressionSelector
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.resolve.ImplLookup
+import org.rust.lang.core.resolve.knownItems
+import org.rust.lang.core.types.TraitRef
+import org.rust.lang.core.types.ty.Ty
+import org.rust.lang.core.types.ty.TyReference
+import org.rust.lang.core.types.ty.TyStr
+import org.rust.lang.core.types.ty.TyUnit
+import org.rust.lang.core.types.type
+
+class PrintlnPostfixTemplate(provider: RsPostfixTemplateProvider, private val macroName: String = "println") :
+    PostfixTemplateWithExpressionSelector(null, macroName, "$macroName!(\"{:?}\", expr);",
+        RsTopMostInScopeSelector { it.isNotIgnored && (it.implementsDebug || it.implementsDisplay) }, provider) {
+
+    private class MacroCreator(private val editor: Editor, private val psiFactory: RsPsiFactory,
+                               macroName: String, isDebug: Boolean) {
+        private val macroStart = "$macroName!(\"{"
+        private val formatter = if (isDebug) ":?" else ""
+
+        fun replaceWithMacro(expression: PsiElement, addTrailingComma: Boolean = false) {
+            val newElement = expression.replace(createMacro(expression.text, expression.parent is RsBlock))
+            if (addTrailingComma) {
+                newElement.add(psiFactory.createComma())
+            }
+            updateSelection(newElement)
+        }
+
+        fun createMacro(expressionText: String, addTrailingSemicolon: Boolean = true): RsExpr {
+            val printlnExpression = psiFactory.createExpression("$macroStart$formatter}\", $expressionText)")
+            if (addTrailingSemicolon) {
+                printlnExpression.add(psiFactory.createSemicolon())
+            }
+            return printlnExpression
+        }
+
+        fun updateSelection(macroElement: PsiElement) {
+            val selectionStart = macroElement.startOffset + macroStart.length
+            val selectionEnd = selectionStart + formatter.length
+            editor.selectionModel.setSelection(selectionStart, selectionEnd)
```

Comment:
I'm not sure whether it should set selection. Is it really useful? In my case, once I've got generated `println!`, I want to press `Enter` and continue writing code. But `setSelection` force me to unselect it to continue.
If you think it is useful, just keep it. Otherwise, consider something like this:
```
        fun updateCaret(macroElement: PsiElement) {
            val offset = if (fmt == Fmt.None) {
                macroElement.endOffset
            } else {
                macroElement.startOffset + macroStart.length + formatter.length
            }
            editor.caretModel.moveToOffset(offset)
        }
```

Reply:
I guess, I'm too used to the awesome `Shift + Enter` shortcut hence did not consider as a problem.
After some thinking I tend to agree with you and it's way more confusing this way, removed the selection completely.

I see a small contradiction between the phrase
> I want to press Enter and continue writing code

and the proposed way to place a caret somewhere around the formatter
> macroElement.startOffset + macroStart.length + formatter.length

so I just place the cared in the end of the println expression in every case.